Dockerfile

```
@@ -0,0 +1,47 @@
+COPY package*.json ./
+# Use --omit=dev to save space and use npm prune later
+RUN npm install --production=false && npm cache clean --force
+
+# Copy application code
+COPY . .
+
+# Build the application
+RUN npm run build
+
+# Prune dependencies to save space
+RUN npm prune --production
```

Hadolint (2.12.0)

[info] Multiple consecutive `RUN` instructions. Consider consolidation.

(DL3059)